diff --git a/CMakeLists.txt b/CMakeLists.txt
index eec0993..0e193ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,11 +30,21 @@ elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL  msp430g2210)
     add_definitions(-DCONFIG_DEBUG_ASSERT=0)
     #...
 elseif(LIB_ONLY)
-    add_definitions(-DCONFIG_DEBUG_PRINTF=0)
-    add_definitions(-DCONFIG_DEBUG_ASSERT=0)
-    add_definitions(-DCONFIG_HAVE_OWN_OFLAGS=1)
     add_definitions(-DCONFIG_HAVE_OWN_ERRNO=0)
+    if (CMAKE_BUILD_TYPE STREQUAL Release)
+        add_definitions(-DCONFIG_DEBUG_PRINTF=0)
+        add_definitions(-DCONFIG_DEBUG_ASSERT=0)
+    else()
+        add_definitions(-DCONFIG_DEBUG_PRINTF=1)
+        add_definitions(-DCONFIG_DEBUG_ASSERT=1)
+    endif()
+    add_definitions(-DCONFIG_HAVE_OWN_ASSERT=0)
     add_definitions(-DCONFIG_BLOCK_DEV_CACHE_SIZE=16)
+    add_definitions(-DCONFIG_EXT4_BLOCKDEVS_COUNT=8)
+    add_definitions(-DCONFIG_EXT4_MOUNTPOINTS_COUNT=8)
+    add_definitions(-DCONFIG_HAVE_OWN_OFLAGS=0)
+    add_definitions(-DCONFIG_UNALIGNED_ACCESS=1)
+    add_definitions(-DCONFIG_USE_USER_MALLOC=0)
 else()
     #Generic example target
     if (WIN32)
@@ -57,7 +67,7 @@ macro(output_configure)
         PROPERTY COMPILE_DEFINITIONS
     )
     file(WRITE
-         ${PROJECT_BINARY_DIR}/include/generated/ext4_config.h
+         ${PROJECT_BINARY_DIR}/include/lwext4/generated/ext4_config.h
          "")
     foreach(item ${definitions})
         string(REGEX MATCH "^CONFIG_" match_res ${item})
@@ -67,15 +77,13 @@ macro(output_configure)
                    "#define ${replace_res} ${CMAKE_MATCH_1}"
                    output_str)
             file(APPEND
-                 ${PROJECT_BINARY_DIR}/include/generated/ext4_config.h
+                 ${PROJECT_BINARY_DIR}/include/lwext4/generated/ext4_config.h
                  "${output_str}\n")
         endif()
     endforeach()
 endmacro()
 output_configure()
 
-add_subdirectory(blockdev)
-
 #Library build
 add_subdirectory(src)
 #Detect all possible warnings for lwext4 target
diff --git a/Makefile b/Makefile
index 12ee5c7..bd5ae43 100644
--- a/Makefile
+++ b/Makefile
@@ -65,7 +65,7 @@ mingw:
 lib_only:
 	rm -R -f build_lib_only
 	mkdir build_lib_only
-	cd build_lib_only && cmake $(COMMON_DEFINITIONS) -DLIB_ONLY=TRUE ..
+	cd build_lib_only && aarch64-none-elf-cmake $(COMMON_DEFINITIONS) -DLIB_ONLY=TRUE ..
 
 all: 
 	generic
diff --git a/include/ext4.h b/include/ext4.h
index 516bbdd..e8db7aa 100644
--- a/include/ext4.h
+++ b/include/ext4.h
@@ -45,13 +45,13 @@ extern "C" {
 #include <stdint.h>
 #include <stddef.h>
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_errno.h>
-#include <ext4_oflags.h>
-#include <ext4_debug.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_errno.h"
+#include "ext4_oflags.h"
+#include "ext4_debug.h"
 
-#include <ext4_blockdev.h>
+#include "ext4_blockdev.h"
 
 /********************************OS LOCK INFERFACE***************************/
 
diff --git a/include/ext4_balloc.h b/include/ext4_balloc.h
index ad481dc..4e2f224 100644
--- a/include/ext4_balloc.h
+++ b/include/ext4_balloc.h
@@ -45,10 +45,10 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
 
-#include <ext4_fs.h>
+#include "ext4_fs.h"
 
 #include <stdint.h>
 #include <stdbool.h>
diff --git a/include/ext4_bcache.h b/include/ext4_bcache.h
index f8bec7d..b7ae438 100644
--- a/include/ext4_bcache.h
+++ b/include/ext4_bcache.h
@@ -41,12 +41,12 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
+#include "ext4_config.h"
 
 #include <stdint.h>
 #include <stdbool.h>
-#include <misc/tree.h>
-#include <misc/queue.h>
+#include "misc/tree.h"
+#include "misc/queue.h"
 
 #define EXT4_BLOCK_ZERO() 	\
 	{.lb_id = 0, .data = 0}
diff --git a/include/ext4_bitmap.h b/include/ext4_bitmap.h
index 6bcb100..cb73b76 100644
--- a/include/ext4_bitmap.h
+++ b/include/ext4_bitmap.h
@@ -41,7 +41,7 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
+#include "ext4_config.h"
 
 #include <stdint.h>
 #include <stdbool.h>
diff --git a/include/ext4_block_group.h b/include/ext4_block_group.h
index 61fbc98..88e00a1 100644
--- a/include/ext4_block_group.h
+++ b/include/ext4_block_group.h
@@ -46,9 +46,9 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_super.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_super.h"
 
 #include <stdint.h>
 #include <stdbool.h>
diff --git a/include/ext4_blockdev.h b/include/ext4_blockdev.h
index 4efbb43..4673479 100644
--- a/include/ext4_blockdev.h
+++ b/include/ext4_blockdev.h
@@ -40,8 +40,8 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_bcache.h>
+#include "ext4_config.h"
+#include "ext4_bcache.h"
 
 #include <stdbool.h>
 #include <stdint.h>
diff --git a/include/ext4_crc32.h b/include/ext4_crc32.h
index a9580c2..3dad1d1 100644
--- a/include/ext4_crc32.h
+++ b/include/ext4_crc32.h
@@ -43,7 +43,7 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
+#include "ext4_config.h"
 
 #include <stdint.h>
 
diff --git a/include/ext4_debug.h b/include/ext4_debug.h
index 10c581e..6eb8aa1 100644
--- a/include/ext4_debug.h
+++ b/include/ext4_debug.h
@@ -41,8 +41,8 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_errno.h>
+#include "ext4_config.h"
+#include "ext4_errno.h"
 
 #if !CONFIG_HAVE_OWN_ASSERT
 #include <assert.h>
diff --git a/include/ext4_dir.h b/include/ext4_dir.h
index 0fa631f..d3b1bf5 100644
--- a/include/ext4_dir.h
+++ b/include/ext4_dir.h
@@ -46,11 +46,11 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_blockdev.h>
-#include <ext4_super.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_misc.h"
+#include "ext4_blockdev.h"
+#include "ext4_super.h"
 
 #include <stdint.h>
 
diff --git a/include/ext4_dir_idx.h b/include/ext4_dir_idx.h
index f1c5b26..f3e83d6 100644
--- a/include/ext4_dir_idx.h
+++ b/include/ext4_dir_idx.h
@@ -46,11 +46,11 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
 
-#include <ext4_fs.h>
-#include <ext4_dir.h>
+#include "ext4_fs.h"
+#include "ext4_dir.h"
 
 #include <stdint.h>
 #include <stdbool.h>
diff --git a/include/ext4_errno.h b/include/ext4_errno.h
index 2d92280..edf89a9 100644
--- a/include/ext4_errno.h
+++ b/include/ext4_errno.h
@@ -40,7 +40,7 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
+#include "ext4_config.h"
 
 #if !CONFIG_HAVE_OWN_ERRNO
 #include <errno.h>
diff --git a/include/ext4_extent.h b/include/ext4_extent.h
index fee0926..0d4e1e3 100644
--- a/include/ext4_extent.h
+++ b/include/ext4_extent.h
@@ -44,9 +44,9 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_inode.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_inode.h"
 
 void ext4_extent_tree_init(struct ext4_inode_ref *inode_ref);
 
diff --git a/include/ext4_fs.h b/include/ext4_fs.h
index c520326..9653f47 100644
--- a/include/ext4_fs.h
+++ b/include/ext4_fs.h
@@ -46,9 +46,9 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_misc.h"
 
 #include <stdint.h>
 #include <stdbool.h>
diff --git a/include/ext4_hash.h b/include/ext4_hash.h
index 43ecbab..17989a1 100644
--- a/include/ext4_hash.h
+++ b/include/ext4_hash.h
@@ -41,7 +41,7 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
+#include "ext4_config.h"
 
 #include <stdint.h>
 
diff --git a/include/ext4_ialloc.h b/include/ext4_ialloc.h
index e845c79..cea3fe6 100644
--- a/include/ext4_ialloc.h
+++ b/include/ext4_ialloc.h
@@ -46,8 +46,8 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
 
 /**@brief Calculate and set checksum of inode bitmap.
  * @param sb superblock pointer.
diff --git a/include/ext4_inode.h b/include/ext4_inode.h
index bee5b8a..e0ca6e3 100644
--- a/include/ext4_inode.h
+++ b/include/ext4_inode.h
@@ -46,8 +46,8 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
 
 #include <stdint.h>
 
diff --git a/include/ext4_journal.h b/include/ext4_journal.h
index 415618b..76a4a87 100644
--- a/include/ext4_journal.h
+++ b/include/ext4_journal.h
@@ -41,10 +41,10 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <misc/queue.h>
-#include <misc/tree.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "misc/queue.h"
+#include "misc/tree.h"
 
 struct jbd_fs {
 	struct ext4_blockdev *bdev;
diff --git a/include/ext4_mbr.h b/include/ext4_mbr.h
index 97a4459..858ab02 100644
--- a/include/ext4_mbr.h
+++ b/include/ext4_mbr.h
@@ -41,8 +41,8 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_blockdev.h>
+#include "ext4_config.h"
+#include "ext4_blockdev.h"
 
 /**@brief Master boot record block devices descriptor*/
 struct ext4_mbr_bdevs {
diff --git a/include/ext4_mkfs.h b/include/ext4_mkfs.h
index aadedb0..16a05f8 100644
--- a/include/ext4_mkfs.h
+++ b/include/ext4_mkfs.h
@@ -41,11 +41,11 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
 
-#include <ext4_blockdev.h>
-#include <ext4_fs.h>
+#include "ext4_blockdev.h"
+#include "ext4_fs.h"
 
 #include <stdbool.h>
 #include <stdint.h>
diff --git a/include/ext4_super.h b/include/ext4_super.h
index 5afd18d..b420225 100644
--- a/include/ext4_super.h
+++ b/include/ext4_super.h
@@ -46,9 +46,9 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_misc.h"
 
 /**@brief   Blocks count get stored in superblock.
  * @param   s superblock descriptor
diff --git a/include/ext4_trans.h b/include/ext4_trans.h
index b17373c..e3cb28a 100644
--- a/include/ext4_trans.h
+++ b/include/ext4_trans.h
@@ -41,8 +41,8 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
 
 
 /**@brief   Mark a buffer dirty and add it to the current transaction.
diff --git a/include/ext4_types.h b/include/ext4_types.h
index c9cdd34..2fd051e 100644
--- a/include/ext4_types.h
+++ b/include/ext4_types.h
@@ -46,9 +46,9 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_blockdev.h>
-#include <misc/tree.h>
+#include "ext4_config.h"
+#include "ext4_blockdev.h"
+#include "misc/tree.h"
 
 #include <stddef.h>
 #include <stdint.h>
diff --git a/include/ext4_xattr.h b/include/ext4_xattr.h
index d79febc..227b3a6 100644
--- a/include/ext4_xattr.h
+++ b/include/ext4_xattr.h
@@ -41,9 +41,9 @@
 extern "C" {
 #endif
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_inode.h>
+#include "ext4_config.h"
+#include "ext4_types.h"
+#include "ext4_inode.h"
 
 struct ext4_xattr_info {
 	uint8_t name_index;
diff --git a/src/ext4.c b/src/ext4.c
index b270361..849cb32 100644
--- a/src/ext4.c
+++ b/src/ext4.c
@@ -34,24 +34,24 @@
  * @brief Ext4 high level operations (file, directory, mountpoints...)
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_oflags.h>
-#include <ext4_debug.h>
-
-#include <ext4.h>
-#include <ext4_trans.h>
-#include <ext4_blockdev.h>
-#include <ext4_fs.h>
-#include <ext4_dir.h>
-#include <ext4_inode.h>
-#include <ext4_super.h>
-#include <ext4_block_group.h>
-#include <ext4_dir_idx.h>
-#include <ext4_xattr.h>
-#include <ext4_journal.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_oflags.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4.h"
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_dir.h"
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_block_group.h"
+#include "lwext4/ext4_dir_idx.h"
+#include "lwext4/ext4_xattr.h"
+#include "lwext4/ext4_journal.h"
 
 
 #include <stdlib.h>
diff --git a/src/ext4_balloc.c b/src/ext4_balloc.c
index 8945554..e0e4d09 100644
--- a/src/ext4_balloc.c
+++ b/src/ext4_balloc.c
@@ -38,20 +38,20 @@
  * @brief Physical block allocator.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_trans.h>
-#include <ext4_balloc.h>
-#include <ext4_super.h>
-#include <ext4_crc32.h>
-#include <ext4_block_group.h>
-#include <ext4_fs.h>
-#include <ext4_bitmap.h>
-#include <ext4_inode.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_balloc.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_block_group.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_bitmap.h"
+#include "lwext4/ext4_inode.h"
 
 /**@brief Compute number of block group from block address.
  * @param sb superblock pointer.
diff --git a/src/ext4_bcache.c b/src/ext4_bcache.c
index 9d3c7fb..26803a9 100644
--- a/src/ext4_bcache.c
+++ b/src/ext4_bcache.c
@@ -34,12 +34,12 @@
  * @brief Block cache allocator.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_bcache.h>
-#include <ext4_blockdev.h>
-#include <ext4_debug.h>
-#include <ext4_errno.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_bcache.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_debug.h"
+#include "lwext4/ext4_errno.h"
 
 #include <string.h>
 #include <stdlib.h>
diff --git a/src/ext4_bitmap.c b/src/ext4_bitmap.c
index 43b6431..baa0eb7 100644
--- a/src/ext4_bitmap.c
+++ b/src/ext4_bitmap.c
@@ -34,13 +34,13 @@
  * @brief Block/inode bitmap allocator.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
-#include <ext4_bitmap.h>
+#include "lwext4/ext4_bitmap.h"
 
 void ext4_bmap_bits_free(uint8_t *bmap, uint32_t sbit, uint32_t bcnt)
 {
diff --git a/src/ext4_block_group.c b/src/ext4_block_group.c
index d2bb1b1..ba63897 100644
--- a/src/ext4_block_group.c
+++ b/src/ext4_block_group.c
@@ -39,13 +39,13 @@
  * @brief Block group function set.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
-#include <ext4_block_group.h>
+#include "lwext4/ext4_block_group.h"
 
 /**@brief CRC-16 look up table*/
 static uint16_t const crc16_tab[256] = {
diff --git a/src/ext4_blockdev.c b/src/ext4_blockdev.c
index c01093a..a044e73 100644
--- a/src/ext4_blockdev.c
+++ b/src/ext4_blockdev.c
@@ -34,15 +34,15 @@
  * @brief Block device module.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_blockdev.h>
-#include <ext4_fs.h>
-#include <ext4_journal.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_journal.h"
 
 #include <string.h>
 #include <stdlib.h>
diff --git a/src/ext4_crc32.c b/src/ext4_crc32.c
index 17ae0d0..43bcd74 100644
--- a/src/ext4_crc32.c
+++ b/src/ext4_crc32.c
@@ -36,13 +36,13 @@
  * @brief Crc32c routine. Taken from FreeBSD kernel.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
-#include "ext4_crc32.h"
+#include "lwext4/ext4_crc32.h"
 
 static const uint32_t crc32_tab[] = {
 	0x00000000, 0x77073096, 0xee0e612c, 0x990951ba, 0x076dc419, 0x706af48f,
diff --git a/src/ext4_debug.c b/src/ext4_debug.c
index 356a157..3858975 100644
--- a/src/ext4_debug.c
+++ b/src/ext4_debug.c
@@ -34,11 +34,11 @@
  * @brief Debug printf and assert macros.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
 #include <stdarg.h>
 
diff --git a/src/ext4_dir.c b/src/ext4_dir.c
index 29a51c5..5638830 100644
--- a/src/ext4_dir.c
+++ b/src/ext4_dir.c
@@ -39,18 +39,18 @@
  * @brief Directory handle procedures.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_trans.h>
-#include <ext4_dir.h>
-#include <ext4_dir_idx.h>
-#include <ext4_crc32.h>
-#include <ext4_inode.h>
-#include <ext4_fs.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_dir.h"
+#include "lwext4/ext4_dir_idx.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_fs.h"
 
 #include <string.h>
 
diff --git a/src/ext4_dir_idx.c b/src/ext4_dir_idx.c
index f7a64dd..0bbe6ea 100644
--- a/src/ext4_dir_idx.c
+++ b/src/ext4_dir_idx.c
@@ -34,21 +34,21 @@
  * @brief Directory indexing procedures.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_trans.h>
-#include <ext4_dir_idx.h>
-#include <ext4_dir.h>
-#include <ext4_blockdev.h>
-#include <ext4_fs.h>
-#include <ext4_super.h>
-#include <ext4_inode.h>
-#include <ext4_crc32.h>
-#include <ext4_hash.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_dir_idx.h"
+#include "lwext4/ext4_dir.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_hash.h"
 
 #include <string.h>
 #include <stdlib.h>
diff --git a/src/ext4_extent.c b/src/ext4_extent.c
index abac59b..4dc4b6f 100644
--- a/src/ext4_extent.c
+++ b/src/ext4_extent.c
@@ -8,19 +8,19 @@
  * of the License, or (at your option) any later version.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_blockdev.h>
-#include <ext4_trans.h>
-#include <ext4_fs.h>
-#include <ext4_super.h>
-#include <ext4_crc32.h>
-#include <ext4_balloc.h>
-#include <ext4_extent.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_balloc.h"
+#include "lwext4/ext4_extent.h"
 
 #include <stdlib.h>
 #include <string.h>
diff --git a/src/ext4_fs.c b/src/ext4_fs.c
index 8e229ef..2f60ddb 100644
--- a/src/ext4_fs.c
+++ b/src/ext4_fs.c
@@ -38,23 +38,23 @@
  * @brief More complex filesystem functions.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_trans.h>
-#include <ext4_fs.h>
-#include <ext4_blockdev.h>
-#include <ext4_super.h>
-#include <ext4_crc32.h>
-#include <ext4_block_group.h>
-#include <ext4_balloc.h>
-#include <ext4_bitmap.h>
-#include <ext4_inode.h>
-#include <ext4_ialloc.h>
-#include <ext4_extent.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_block_group.h"
+#include "lwext4/ext4_balloc.h"
+#include "lwext4/ext4_bitmap.h"
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_ialloc.h"
+#include "lwext4/ext4_extent.h"
 
 #include <string.h>
 
diff --git a/src/ext4_hash.c b/src/ext4_hash.c
index ff6d031..e8be8d2 100644
--- a/src/ext4_hash.c
+++ b/src/ext4_hash.c
@@ -61,11 +61,11 @@
  * @brief Directory indexing hash functions.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
 #include <string.h>
 
diff --git a/src/ext4_ialloc.c b/src/ext4_ialloc.c
index f2c796f..53f3480 100644
--- a/src/ext4_ialloc.c
+++ b/src/ext4_ialloc.c
@@ -39,20 +39,20 @@
  * @brief Inode allocation procedures.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_trans.h>
-#include <ext4_ialloc.h>
-#include <ext4_super.h>
-#include <ext4_crc32.h>
-#include <ext4_fs.h>
-#include <ext4_blockdev.h>
-#include <ext4_block_group.h>
-#include <ext4_bitmap.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_ialloc.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_block_group.h"
+#include "lwext4/ext4_bitmap.h"
 
 /**@brief  Convert i-node number to relative index in block group.
  * @param sb    Superblock
diff --git a/src/ext4_inode.c b/src/ext4_inode.c
index ff3c234..a348a8d 100644
--- a/src/ext4_inode.c
+++ b/src/ext4_inode.c
@@ -39,14 +39,14 @@
  * @brief Inode handle functions
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_inode.h>
-#include <ext4_super.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_super.h"
 
 /**@brief  Compute number of bits for block count.
  * @param block_size Filesystem block_size
diff --git a/src/ext4_journal.c b/src/ext4_journal.c
index 9d998a1..5cb7da3 100644
--- a/src/ext4_journal.c
+++ b/src/ext4_journal.c
@@ -35,18 +35,18 @@
  * @brief Journal handle functions
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_fs.h>
-#include <ext4_super.h>
-#include <ext4_journal.h>
-#include <ext4_blockdev.h>
-#include <ext4_crc32.h>
-#include <ext4_journal.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_journal.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_journal.h"
 
 #include <string.h>
 #include <stdlib.h>
diff --git a/src/ext4_mbr.c b/src/ext4_mbr.c
index 0376545..14b64db 100644
--- a/src/ext4_mbr.c
+++ b/src/ext4_mbr.c
@@ -34,13 +34,13 @@
  * @brief Master boot record parser
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
-#include <ext4_mbr.h>
+#include "lwext4/ext4_mbr.h"
 
 #include <inttypes.h>
 #include <string.h>
diff --git a/src/ext4_mkfs.c b/src/ext4_mkfs.c
index 0dfc91f..32db434 100644
--- a/src/ext4_mkfs.c
+++ b/src/ext4_mkfs.c
@@ -34,20 +34,20 @@
  * @brief
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_super.h>
-#include <ext4_block_group.h>
-#include <ext4_dir.h>
-#include <ext4_dir_idx.h>
-#include <ext4_fs.h>
-#include <ext4_inode.h>
-#include <ext4_ialloc.h>
-#include <ext4_mkfs.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_block_group.h"
+#include "lwext4/ext4_dir.h"
+#include "lwext4/ext4_dir_idx.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_ialloc.h"
+#include "lwext4/ext4_mkfs.h"
 
 #include <inttypes.h>
 #include <string.h>
diff --git a/src/ext4_super.c b/src/ext4_super.c
index 092c38b..a137a76 100644
--- a/src/ext4_super.c
+++ b/src/ext4_super.c
@@ -39,14 +39,14 @@
  * @brief Superblock operations.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
-
-#include <ext4_super.h>
-#include <ext4_crc32.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
+
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_crc32.h"
 
 uint32_t ext4_block_group_cnt(struct ext4_sblock *s)
 {
diff --git a/src/ext4_trans.c b/src/ext4_trans.c
index f228751..973b3d3 100644
--- a/src/ext4_trans.c
+++ b/src/ext4_trans.c
@@ -35,14 +35,14 @@
  * @brief Ext4 transaction buffer operations.
  */
 
-#include <ext4_config.h>
-#include <ext4_types.h>
-#include <ext4_misc.h>
-#include <ext4_errno.h>
-#include <ext4_debug.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_types.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_debug.h"
 
-#include <ext4_fs.h>
-#include <ext4_journal.h>
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_journal.h"
 
 int ext4_trans_set_block_dirty(struct ext4_buf *buf)
 {
diff --git a/src/ext4_xattr.c b/src/ext4_xattr.c
index f8a5778..a3c5dc0 100644
--- a/src/ext4_xattr.c
+++ b/src/ext4_xattr.c
@@ -16,21 +16,21 @@
  * @brief Extended Attribute manipulation.
  */
 
-#include <ext4_config.h>
-#include <ext4_debug.h>
-#include <ext4_errno.h>
-#include <ext4_misc.h>
-#include <ext4_types.h>
-
-#include <ext4_balloc.h>
-#include <ext4_block_group.h>
-#include <ext4_blockdev.h>
-#include <ext4_crc32.h>
-#include <ext4_fs.h>
-#include <ext4_inode.h>
-#include <ext4_super.h>
-#include <ext4_trans.h>
-#include <ext4_xattr.h>
+#include "lwext4/ext4_config.h"
+#include "lwext4/ext4_debug.h"
+#include "lwext4/ext4_errno.h"
+#include "lwext4/ext4_misc.h"
+#include "lwext4/ext4_types.h"
+
+#include "lwext4/ext4_balloc.h"
+#include "lwext4/ext4_block_group.h"
+#include "lwext4/ext4_blockdev.h"
+#include "lwext4/ext4_crc32.h"
+#include "lwext4/ext4_fs.h"
+#include "lwext4/ext4_inode.h"
+#include "lwext4/ext4_super.h"
+#include "lwext4/ext4_trans.h"
+#include "lwext4/ext4_xattr.h"
 
 #include <stdlib.h>
 #include <string.h>
